### DISCUSSION TITLE
not getting response from server on some specific inputs

### DISCUSSION THREAD

**Discussion by shubham-gupta0** [fastapi/fastapi#11749] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import os
import tempfile
from fastapi import FastAPI, File, UploadFile, HTTPException
from fastapi.responses import FileResponse
from ultralytics import YOLO
from PIL import Image, ImageDraw
import numpy as np

app = FastAPI()

# Load your YOLO model (adjust the path as needed)
model_path = "best.pt"
model = YOLO(model_path, task="detect")

def draw_boxes(image: Image.Image, boxes):
    """Draw bounding boxes on the image."""
    draw = ImageDraw.Draw(image)
    for box in boxes:
        x1, y1, x2, y2 = box[:4]
        draw.rectangle([x1, y1, x2, y2], outline="red", width=3)
    return image

@app.get("/")
async def hello_world():
    return {"message": "Hello, World!"}

@app.post("/upload")
def upload_image(file: UploadFile = File(...)):
    try:
        # Read the image file
        image = Image.open(file.file)

        # Convert PIL image to numpy array
        image_np = np.array(image)

        # Perform inference
        results = model(image_np, conf=0.5, imgsz=640, iou=0.25)
        
        # Process results
        boxes = results[0].boxes.xyxy  # Bounding boxes

        # Check if there are no bounding boxes
        if len(boxes) == 0:
            raise HTTPException(status_code=400, detail="No objects detected in the image.")

        # Draw bounding boxes on the image
        image_with_boxes = draw_boxes(image, boxes)

        # Save the image with bounding boxes to a temporary file
        temp_file = tempfile.NamedTemporaryFile(delete=False, suffix=".jpg")
        image_with_boxes.save(temp_file, format="JPEG")
        temp_file.close()

        # Serve the file
        return FileResponse(temp_file.name, media_type="image/jpeg", filename="image_with_boxes.jpg")

    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
    finally:
        # Clean up the temporary file after response is sent
        @app.on_event("shutdown")
        def remove_temp_file():
            try:
                os.remove(temp_file.name)
            except Exception:
                pass

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=3000)
```

### Description

`i created a api using fastapi i takes a image input processes it through the model and return the image with the bounding box drawn on image. want to integrate it in flutter app tested it locally it was working fine.then i hosted it on ubuntu(20.04) server(16GB RAM 8cores cpu) using nginx and pm2.works fine on some images but doesnot give response on some specific image and doesn't even process it.

the below is nginx configuration: server
 {
 listen 80;

    server_name server_ip example.com;

location / {
    proxy_pass http://localhost:3000;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
}

}

has replaced the server_ip with the ip of server and example.com with the url

tried the api with the python script to test so got few errors

File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\urllib3\connectionpool.py",line 537, in _make_request
    response = conn.getresponse()
               ^^^^^^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\urllib3\connection.py", line 466, in getresponse
    httplib_response = super().getresponse()
                       ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\http\client.py", line 1423, in getresponse
    response.begin()
  File "C:\Python312\Lib\http\client.py", line 331, in begin
    version, status, reason = self._read_status()
                              ^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\http\client.py", line 292, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\socket.py", line 707, in readinto
    return self._sock.recv_into(b)
           ^^^^^^^^^^^^^^^^^^^^^^^
TimeoutError: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\requests\adapters.py", line 667, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\urllib3\connectionpool.py", line 847, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\urllib3\util\retry.py", line 470, in increment
    raise reraise(type(error), error, _stacktrace)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\urllib3\util\util.py", line 39, in reraise
    raise value
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\urllib3\connectionpool.py", line 793, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\urllib3\connectionpool.py", line 539, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\urllib3\connectionpool.py", line 370, in _raise_timeout
    raise ReadTimeoutError(
urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='14.139.54.210', port=3000): Read timed out. (read timeout=None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\nandg\OneDrive\Desktop\python\fastapitest.py", line 7, in <module>
    response = requests.post(url, files={"file": file})
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\requests\api.py", line 115, in post
    return request("post", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\requests\sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^`your text``
`
 # File "C:\Users\nandg\OneDrive\Desktop\python\.venv\Lib\site-packages\requests\adapters.py", line 713, in #send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPConnectionPool(host='14.139.54.210', port=3000): Read timed out. (read timeout=None)`

### Operating System

Other

### Operating System Details

ubuntu(20.04)

### FastAPI Version

0.111.0

### Pydantic Version

2,7,4

### Python Version

3.8.10

**Comment by chyok** [comment]
Hi @shubham-gupta0 ,
This looks more like an Nginx configuration file issue rather than a FastAPI problem. You can try using `telnet` to test if the remote host's port 3000 is accessible, and gradually investigate the issue step by step.

**Reply by shubham-gupta0** [reply]
i tried it testing but for some inputs it is responding and also returning the output but for some images it is not responding

**Comment by shubham-gupta0** [comment]
i checked my pm2 logs the following is the error i am facing :


0|fast_api | [2024-06-21 14:28:09 +0530] [293114] [INFO] Waiting for application startup.
0|fast_api | [2024-06-21 14:28:09 +0530] [293114] [INFO] Application startup complete.
0|fast_api | [2024-06-22 00:31:47 +0530] [293115] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-22 00:31:47 +0530] [293114] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-22 10:06:07 +0530] [293114] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-22 11:08:26 +0530] [293115] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-22 11:12:21 +0530] [293114] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-22 23:04:53 +0530] [293114] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-22 23:04:53 +0530] [293115] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-23 05:44:48 +0530] [293116] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-23 06:51:04 +0530] [293113] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-23 06:51:13 +0530] [293116] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-23 08:40:19 +0530] [293115] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-23 09:46:08 +0530] [293115] [WARNING] Invalid HTTP request received.
0|fast_api | [2024-06-23 09:46:17 +0530] [293113] [WARNING] Invalid HTTP request received.

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussion clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
